spec/lib/api/2.0/views-spec.js

```diff
@@ -0,0 +1,147 @@
+// Copyright 2015, EMC, Inc.
+/* jshint node:true */
+
+'use strict';
+
+describe('Http.Api.Views', function () {
+    var viewsProtocol;
+    var _;
+
+    before('start HTTP server', function () {
+        this.timeout(10000);
+        viewsProtocol = {
+            load: sinon.stub(),
+            getAll: sinon.stub(),
+            get: sinon.stub(),
+            put: sinon.stub(),
+            unlink: sinon.stub()
+        };
+
+        return helper.startServer([
+            dihelper.simpleWrapper(viewsProtocol, 'Views')
+        ]).then(function() {
+            _ = helper.injector.get('_');
+        });
+    });
+
+    beforeEach('reset stubs', function () {
+        _(viewsProtocol).methods().forEach(function (method) {
```

Comment:
@brianparry I think doing it the way you're doing it is valid, but let me add another argument in favor of the sandbox approach (as a general approach, not asking you to change anything):

The `methods().forEach((method) => { method.reset() })` approach is vulnerable to someone doing: `method.restore()` in their test, which is something I commonly do when I need the real version of a method so I can test its code path. Also I think that it ends up being easier to reason about what is being stubbed and what is not in a unit test when the test starts with something like:

```
this.sandbox.stub(myClass, 'myMethod1');
this.sandbox.stub(myClass, 'myMethod2');
```

Because of this, it's defined in the unit test what is stubbed, rather than having to refer to the one or more before blocks that meta-program a bunch of stubs (and usually not directly by method name, but by iteration). With the `methods().forEach()` approach, I think it's actually much harder to track down what is a stub and what is not, because `viewsProtocol[method]` doesn't tell me at first glance that `viewsProtocol.myMethod` is stubbed. This can get even harder when there are multiple describe and beforeEach blocks doing things like this to different objects.